File: python/aoc2022/days/day10.py
def day10():
    with open('../input/input10.txt', 'r') as f:
        data = f.read().splitlines()

    x = [1, 1]

    for instruction in data:
        if instruction[0] == 'n':
            x.append(x[-1])
        else:
            val = int(instruction.split(' ')[1])
            x.append(x[-1])
            x.append(val + x[-1])

    print('Part 1:', sum([x[i] * i for i in [20, 60, 100, 140, 180, 220]]))

    display = ''
    for row in range(6):
        display += '\n'
        pos = 0
        for cycle in range(40 * row + 1, 40*(row + 1) + 1):
            sprite = x[cycle]
            if sprite - 1 <= pos <= sprite + 1:
                display += '#'
            else:
                display += '.'
            pos += 1

    print('Part 2:', display)

File: rust/src/aoc2022/day10.rs
use rdcl_aoc_helpers::input::WithReadLines;
use std::fs::File;

pub fn day10(path: &String) {
    println!("Day 10");
    let data:Vec<String> = File::open(path).read_lines::<String>(1).collect();

    let mut x:Vec<i32> = vec![1, 1];

    for instr in data{
        if instr.chars().nth(0).unwrap() == 'n' {
            x.push(*x.last().unwrap());
        } else {
            let val:i32 = instr.split(' ').nth(1).unwrap().parse::<i32>().unwrap();
            x.push(*x.last().unwrap());
            x.push(*x.last().unwrap() + val);
        }
    }
    println!("Part 1: {}", (20..221).step_by(40).map(|i| (i as i32)*x[i]).sum::<i32>());

    let mut s = String::new();
    for row in 0..6 {
        s += "\n";
        let mut pos = 0;
        for cycle in (40*row+1)..(40*(row+1)+1) {
            let sprite = x[cycle];
            if sprite - 1 <= pos && pos <= sprite + 1 {
                s+="#";
            } else {
                s+=".";
            }
            pos += 1;
        }
    }
    println!("Part 2: {}", s);
}